kornia/contrib/sam/predictor.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import annotations
+
+from typing import Any
+
+import torch
+
+from kornia.augmentation import AugmentationSequential, LongestMaxSize
+from kornia.core import Device, Tensor
+from kornia.core.check import KORNIA_CHECK_SHAPE
+from kornia.geometry.boxes import Boxes
+from kornia.geometry.keypoints import Keypoints
+
+from .architecture import Sam
+from .model import SamPrediction
+
+
+class SamPredictor:
+    """Uses SAM to calculate the image embedding for an image, and then allow repeated, efficient mask prediction
+    given prompts.
+
+    Args:
+        sam_model: The model to use for mask prediction.
+    """
+
+    def __init__(self, sam_model: Sam) -> None:
+        super().__init__()
+        self.model = sam_model
+        self._longside_size = self.model.image_encoder.img_size
+        transforms = (LongestMaxSize(self._longside_size),)
+        self.tfs = AugmentationSequential(*transforms, same_on_batch=True)
+
+    def device(self) -> Device:
+        return self.model.device
+
+    @torch.no_grad()
+    def __call__(
+        self,
+        image: Tensor,
+        point_coords: Keypoints | None = None,
+        point_labels: Tensor | None = None,
+        boxes: Boxes | None = None,
```

Comment:
Should it be more flexible and also support raw tensors for the points and boxes here?

Reply:
Yes. it is better to make it work for both tensor type and our own data types.

Reply:
+1

Reply:
Also I’m more fan to have separated calls to predict each of the types